File: contracts/rust/dummy/dummy.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package dummy

import "github.com/iotaledger/wasp/packages/vm/wasmlib"

// fails with error if failInitParam exists
func funcInit(ctx *wasmlib.ScFuncContext, params *FuncInitParams) {
	if params.FailInitParam.Exists() {
		ctx.Panic("dummy: failing on purpose")
	}
}

File: contracts/rust/dummy/src/dummy.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use wasmlib::*;

use crate::*;

pub fn func_init(ctx: &ScFuncContext, params: &FuncInitParams) {
    if params.fail_init_param.exists() {
        ctx.panic("dummy: failing on purpose");
    }
}
